src/postamble.js

```diff
@@ -363,6 +363,16 @@ function checkUnflushedContent() {
 
 /** @param {boolean|number=} implicit */
 function exit(status, implicit) {
+#if USE_PTHREADS
+  if (ENVIRONMENT_IS_PTHREAD) {
+    // When running in a pthread we propagate the exit out the worker.js where
+    // it gets tranlated in an 'exitProcess' message on the main thread.
+    // This allows the main thread to decide if (or if not) it wants to bring
+    // down the entire application, based on the main thread's noExitRuntime.
+    throw new ExitStatus(status);
```

Comment:
Is this enough? The try-catch that gets this is on the main invocation of the pthread's main function. But if the pthread set a main loop, it will be called asynchronously. I think we can just send the postMessage from here. Then the thrown error is just to halt execution in the pthread.

Reply:
You are right that sounds better.  Done.